### PR TITLE
split emails by number of attachments

### PR DESCRIPTION
**Pull request by asmega**
AWS SES is only capable of sending 10MB per email
Hence when there are many attachments there is the possibility this
threshold is passed
To workaround this the emails are split based on the number of
attachments
As attachments are Base64 encoded there is an overhead and once
attachments reach ~7.5MB there is risk the message will become too large

### CODE REVIEW

File: spec/services/process_submission_service_spec.rb [hunk 1]
```diff
@@ -0,0 +1,483 @@
...
+
+    context 'given a valid submission_id' do
+      let(:submission_id) { submission.id }
+
+      before do
+        allow(Submission).to receive(:find).with(submission_id).and_return(submission)
+      end
+
+      it 'loads the Submission' do
+        expect(Submission).to receive(:find).with(submission_id).and_return(submission)
```
Comment (tommotaylor) [inline spec/services/process_submission_service_spec.rb]: Line 99 and Line 103 just test each other. Line 99 will blow up if it doesn't receive find with the submission id, so I would say that either the L102 spec doesn't need to exist or we can put the `before`/`do` block after L102 in its own context.

File: app/services/process_submission_service.rb [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+          response = EmailService.send_mail(
+            from:         mail.from,
+            to:           mail.to,
+            subject:      "#{mail.subject} {#{submission_id}} [#{n+1}/#{number_of_attachments(mail)}]",
+            body_parts:   retrieve_mail_body_parts(mail),
+            attachments:  [a]
+          )
+
+          submission.responses << response.to_h
+        end
```
[the remark this answers is not in the document]
Comment (asmega) [reply app/services/process_submission_service.rb]: citizen facing users will only receive one attachment at most so we keep the old/existing behaviour. so this is for backwards compatibility. i don't think we should suddenly burden them with a random guid or supply `[1/1]` which is superfluous information

File: spec/services/process_submission_service_spec.rb (hunk 1 above)
Comment (asmega) [reply spec/services/process_submission_service_spec.rb]: I've made a slight tweak and removed the extra test

**Comment by asmega** [comment]
I wrote a lot of the extra tests that you see. The original tests were testing implementation details and not the public api. I had to follow in the same vein in order to gain any confidence of making any changes to this overly responsible class. But I agree this class still needs splitting up.

File: app/services/process_submission_service.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (solidgoldpig) [reply app/services/process_submission_service.rb]: And also the citizen-facing part should (eventually) be handled by Notify…

**Pull request merged**